Updated contributing and readme file

n1rjal opened the pull request:
I have updated the readme and styles.md file. Please go through this


@Sailesh-Singh can you verify spelling and grammatical issues
@rex9840 please look into if I should add something here.

Thanks.

n1rjal committed:
feat: Updated docs

Added contributing.md, style_guide.md
Updated readme.md

n1rjal committed:
fix: contributing and readme

some missed 404 are removed

n1rjal committed:
fix: contributing readme updated

Sailesh-Singh commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,119 @@
+## How to Contribute
+
+You can contribute in several ways:
+- Reporting bugs
```
Add link to issue template of bug report
Add link to issue template of feature request or enhancement

n1rjal replied:
I think we will need the users to show 
1. Issue Raising page not to template
2. Contributing page, which is this one so no need.

Sailesh-Singh commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,119 @@
+
+## Getting Started
+
+1. **Fork the repository:**
```
Add below link to make forking the repository more convenient 
[Click here](https://github.com/GNOME-Nepal/site-server/fork) to fork.

Sailesh-Singh commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,119 @@
+   ```
+
+3. Setup env
+You can setup env by copying the env.example file and updating its entry to match your local settings.
````
Set up environment variables `env` by copying the `env.example` file and updating its entries to match your local settings.

Sailesh-Singh commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,119 @@
+## Development Workflow
+
+1. **Keep your fork up to date:**
+You can sync the fork using this way in github's UI
```
Make bit tweak
1. **Keep your fork up to date:**
   You can sync the fork using GitHub's UI.
   ![Sync the fork using GitHub's UI](assets/sync_the_fork.png)

Sailesh-Singh commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,119 @@
+   Implement your changes in the relevant files.
+
+3. **Test your changes:**
+    Ensure your changes work as intended and do not introduce new issues. Make sure that the docker image thus created is runnable and doesn't create furthure issues furing build or any production cases.
```
docker -> Docker
create -> cause
furthure -> further
furing-> during
any production cases-> in production.

Sailesh-Singh commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,119 @@
+    Ensure your changes work as intended and do not introduce new issues. Make sure that the docker image thus created is runnable and doesn't create furthure issues furing build or any production cases.
+
+4. **Lint your changes:**
+    Inside the virtual environment, use the following commands
```
Add appropriate comments:
commands-> commands:
   ```bash
   python -m black .  # Formats your code to meet our coding standards
   python -m flake8   # Identifies unused variables and linting issues
   ```

Sailesh-Singh commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,119 @@
+    python -m flake8 # this will tell you about unused variables and linitng issues that you will need to improve.
+    ```
+
+    Please note that our workflow will ignore your changes if it doesn't run or linting is missing. If you want any help with that, please create a PR and then ask the help from maintainer.
````
Correct sentences semantically
if it doesn't run or linting-> if they do not pass or if linting 
 want any help-> need help
 please-> feel free to
 and ~~then~~ ask for ~~the~~ help from maintainer

Sailesh-Singh commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,119 @@
+    Please note that our workflow will ignore your changes if it doesn't run or linting is missing. If you want any help with that, please create a PR and then ask the help from maintainer.
+
+
+4. **Commit your changes:**
```
correct redundant 4 and numbering

Sailesh-Singh commented on STYLE_GUIDE.md:
```diff
@@ -0,0 +1,85 @@
+- [Code Formatting](#code-formatting)
+
+
+## Code Architecture
```
Code Architecture is more about how the project is organized overall and might help onboard new developers, it fits better in the main readme as previous one

n1rjal replied:
But where is code formatting file in our current repository ?

Sailesh-Singh replied:
No, I only asked for Code Architecture block to be placed in readme rather in STYLE_GUIDE.md

Sailesh-Singh commented on STYLE_GUIDE.md:
```diff
@@ -0,0 +1,85 @@
+- Use meaningful variable and function names.
+- Adhere to the language-specific best practices.
+- Ensure your code is modular and reusable.
+- Avoid code duplication.
```
also add "and hardcoding values"
- Include docstrings for all functions, classes, and modules.

Sailesh-Singh commented on STYLE_GUIDE.md:
```diff
@@ -0,0 +1,85 @@
+- Use the imperative mood ("Fix bug" not "Fixed bug").
+- Keep the subject line (first line) under 50 characters.
+- Separate subject from body with a blank line.
+- Provide a detailed description of the changes in the body.
```
add at the end of line : , explaining **what** and **why**

n1rjal replied:
Explain me this with an example. I don't think I can explain these general rules on why

Sailesh-Singh replied:
subject line :- feat: Add user authentication system

Body: What(Added user login and registration features using Django’s built-in authentication system.
The feature includes:
- Login page
- Registration page
- Password reset functionality)
--explains what was done in more detail. It describes the feature and its components.

Body:why(The reason for adding this feature is to allow users to authenticate and access personalized content. This feature is crucial for security and user experience.)
-- explains why the feature was added, providing context for the decision to add user authentication.

Sailesh-Singh commented on STYLE_GUIDE.md:
```diff
@@ -0,0 +1,85 @@
+
+## Branch Naming Conventions
+
+Branch names should be descriptive and follow a consistent pattern. Use the following conventions for naming branches:
```
~~ for naming branches~~
also add 
- Optionally, include an issue or ticket number for tracking (e.g., `feat/issue-55-add-auth`).

Sailesh-Singh commented on STYLE_GUIDE.md:
````diff
@@ -0,0 +1,85 @@
+```
+
+## Code Formatting
+
````
instead of writing in in paragraph break the context into points and sub-points for clear understanding

correction:
Consistent code formatting improves readability and maintainability. Follow these rules:

Use [black](https://black.readthedocs.io/en/stable/) for automatic code formatting.
   -Default configuration with `max_line_length=79` and double quotes.
  -Target Python 3.11 compatibility.

Use [flake8](https://flake8.pycqa.org/) for linting to ensure code quality.
   -Run `flake8` to catch unused variables and other issues.
   -Check for adherence to PEP 8 and best practices.

Before pushing your code, run the following commansd:
python3 -m black .  # Formats code
python3 -m flake8   # Runs lint checks

Sailesh-Singh commented on readme.md:
```diff
@@ -111,12 +111,6 @@ As part of the GNOME Nepal backend server, we are planning to incorporate the fo
 - [ ] **Member Lookup**: Ability for staff contributors to look up members.
 - [ ] **Blog Submission**: Contributors can submit blogs, with super admins verifying content.
 
-## Code Architecture
```
Keep this block in readme

Sailesh-Singh reviewed (changes requested):
Check all suggestions

rex9840 commented:
as  pointed out in review just make the changes as sailesh suggested and you are good to go

n1rjal committed:
feat(gitginore): git ignored ds store

n1rjal committed:
fix(gitignore): fix DS_Store

n1rjal committed:
fix: Updated changes according to pr request

Updated readme.md, style_guide.md and contributing.md according to
changes requested in readme.md
